### PR TITLE
feat: Added interactive-clap for cargo-near

### PR DESCRIPTION
**Pull request by FroVolod**

### CODE REVIEW

File: cargo-near/src/lib.rs [hunk 1]
```diff
@@ -1,136 +0,0 @@
-use std::{env, str::FromStr};
```
Comment (frol) [inline cargo-near/src/lib.rs:1]: lib.rs should stay as we want to make use of cargo-near commands in near-workspaces-rs, similarly to how we use near-cli-rs here. Please, move most of things except main() function back from main.rs to here.

File: cargo-near/src/main.rs [hunk 2]
```diff
@@ -1,27 +1,106 @@
...
+    loop {
+        match <Cmd as interactive_clap::FromCli>::from_cli(
+            Some(cli.clone()),
+            global_context.clone(),
+        ) {
+            interactive_clap::ResultFromCli::Ok(cli_cmd)
+            | interactive_clap::ResultFromCli::Cancel(Some(cli_cmd)) => {
+                eprintln!(
+                    "Your console command:\n{} {}",
+                    std::env::args().next().as_deref().unwrap_or("./cargo-near"),
```
Comment (frol) [inline cargo-near/src/main.rs:84]: We need to make sure that if users call `cargo near build` we don't suggest them to use `cargo-near near build`. I feel it is currently the case, correct me if I am wrong

File: cargo-near/src/abi_command/mod.rs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    /// Include rustdocs in the ABI file
+    #[interactive_clap(long)]
+    pub doc: bool,
+    /// Generate compact (minified) JSON
+    #[interactive_clap(long)]
+    pub compact_abi: bool,
+    /// Copy final artifacts to this directory
+    #[interactive_clap(long)]
+    #[interactive_clap(skip_default_input_arg)]
+    pub out_dir: Option<crate::types::utf8_path_buf::Utf8PathBuf>,
```
Comment (frol) [inline cargo-near/src/abi_command/mod.rs:16]: I don't know the reason behind the use of Utf8PathBuf, so I would love to try and see if PathBuf would work just fine ([we already have it near-cli-rs](https://github.com/near/near-cli-rs/blob/master/src/types/path_buf.rs))

P.S. Obviously, try applying it everywhere we currently have this Utf8PathBuf and if everything works fine, remove the dependency from Cargo.toml as well.

**Review by frol — COMMENTED** [review]
Nice! We are on the right path to getting `cargo near deploy`!

File: cargo-near/src/lib.rs (hunk 1 above)
Comment (FroVolod) [reply cargo-near/src/lib.rs:1]: Done

File: cargo-near/src/main.rs (hunk 2 above)
Comment (FroVolod) [reply cargo-near/src/main.rs]: Done

**Review by frol — CHANGES_REQUESTED** [review]
Please, refactor the project to have a similar structure to near-cli-rs / bos-cli-rs

**Review by frol — APPROVED** [review]
I believe it is ready to replace the previous implementation!

**Pull request merged**